static/templates/actors/character/tabs/spellcasting.hbs

```diff
@@ -66,7 +66,11 @@
                         <section class="spell-ability">
                             <span class="spell-tradition">{{localize entry.statistic.label}}</span>
                             <div class="spellcasting-prof button-group">
-                                <select class="skill-proficiency pf-rank adjust-item-stat-select" data-item-property="system.proficiency.value" data-rank="{{entry.statistic.rank}}" data-dtype="Number">
+                                <select
+                                    class="skill-proficiency pf-rank"
+                                    data-action="{{#if entry.usesSpellProficiency}}update-spellcasting-rank{{else}}update-spellcasting-entry{{/if}}"
```

Comment:
I think you can avoid a specific handler for the regular update if you use data-item-property